### ISSUE TITLE
How to remove a node?

### ISSUE THREAD

**Issue by trusktr** [infamous/famin#1]
Is there a way to remove a node and add it to another parent node (while keeping the DOMElement content intact)?

**Comment by gadicc** [comment]
I'm uh, not that far yet :>  Was mostly focusing on rendering performance until now.

I don't foresee any problems with this, although we'd need to be careful with API naming... e.g. maybe `.detach()` in this case to avoid completely destroying the node... jQuery uses the same API:

> The .detach() method is the same as .remove(), except that .detach() keeps all jQuery data associated with the removed elements. This method is useful when removed elements are to be reinserted into the DOM at a later time.

As always, always great to hear your use case for this to keep in mind.  Can I assume you'd want to do some transition and then on the complete callback, change the node's position in the tree?  (Or the other way around?)

**Comment by trusktr** [comment]
> I assume you'd want to do some transition and then on the complete callback, change the node's position in the tree?

Yep, you guessed it!

Plus I'm sure users will have other use cases, so it should just work. In Andrew's infamous/motor, I made it so that calling `addChild(child)` on a node remove the child from it's parent before adding to node we're calling addChild on (so it's similar to an HTMLElement's appendChild method).
